### ISSUE TITLE
Remove Dots from Component Names

### ISSUE THREAD

**Issue by maximilianweidenauer** [sibvisions/reactUI#278]
It is possible that the server sends component names with dots in them. This is leading to problems because the ReactUI is using the component name as element ids. PrimeReact and I use the `querySelector` function which will interpret the dots in the id as the start of a className which results in errors.

For now I'm removing the dots on client-side. But as discussed on friday this should happen on server side.

**Comment by rjahn** [comment]
Do you have an example?

**Comment by maximilianweidenauer** [comment]
Yes I think this happened while I was coding for V2... The "...About" screens components had "About..." in their name which caused a crash.
![image](https://user-images.githubusercontent.com/66732863/167820377-29d40587-0293-4df9-973e-6070149cd8b8.png)

**Comment by rjahn** [comment]
thank you!

**Comment by rjahn** [comment]
Please verify with latest version in repo.

**Comment by maximilianweidenauer** [comment]
Doesn't seem to work, I'm on the latest version...
![image](https://user-images.githubusercontent.com/66732863/169808249-e8bbf4cb-9ba4-4c69-93af-4ba92ae53671.png)

**Comment by rjahn** [comment]
ok, the used Java method works like defined: https://electrictoolbox.com/valid-characters-html-id-attribute/

Sure, I could remove some characters, but what's allowed for react?

**Comment by maximilianweidenauer** [comment]
I think this sums it up pretty well: https://stackoverflow.com/a/79022

**Comment by maximilianweidenauer** [comment]
The "...About" frame in full transferType mode works now.

Just to be clear :
> Non letters only if name starts with a letter.

You mean that non letters aren't allowed at the start eg. numbers at the start. 
Because currently if the name starts with a number, I add an underscore in front of it

**Comment by rjahn** [comment]
Examples:

```
1234ABC456 -> ABC456
_ABC-_ -> ABC-_
aAbC124 -> aAbC124
```

It's very restrictive, but it's only the style info.

**Comment by maximilianweidenauer** [comment]
Ok thats perfect, then I don't have to worry about that on the client anymore!

**Comment by rjahn** [comment]
Yes, there shouldn't be specific handling on client side.
